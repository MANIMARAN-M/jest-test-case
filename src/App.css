* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}
.App {
  display: flex;
  justify-content: center;
  align-items: center;
}
.LoginForm {
  width: 600px;
  margin: 40px auto;
  background-color: ghostwhite;
  padding: 50px;
}
.LoginForm h1 {
  text-align: center;
  margin-bottom: 20px;
}
.LoginForm button {
  width: 100%;
  padding: 8px;
  margin-top: 20px;
  cursor: pointer;
}
.LoginInoutGroup {
  margin: 10px 0;
}
.LoginInoutGroup input {
  width: 100%;
  padding: 5px;
  outline: none;
}
.testButton {
  padding: 8px 20px;
}
.appButtonTest {
  text-align: center;
  margin: 30px 0;
}
