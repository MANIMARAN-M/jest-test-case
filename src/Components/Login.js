import React, { useState } from "react";

const Login = ({ setisLogged }) => {
  const [LoginFrom, setLoginFrom] = useState({
    MailId: "maran@gmail.com",
    Password: "maran001",
  });
  const [email, setemail] = useState("");
  const [password, setpassword] = useState("");
  //   const [LoginUser, setLoginUser] = useState({
  //     MailId: "",
  //     Password: "",
  //   });
  const SumbmitHandler = (e) => {
    e.preventDefault();
    if (LoginFrom.MailId === email && LoginFrom.Password === password) {
      setLoginFrom({
        ...LoginFrom,
      });
      setisLogged(true);
      setemail("");
      setpassword("");
    } else {
      alert("You Email or Password was wrong!");
      setemail("");
      setpassword("");
    }
  };
  return (
    <form className="LoginForm" id="Form" onSubmit={(e) => SumbmitHandler(e)}>
      <h1>signIn Form</h1>
      <div className="LoginInoutGroup">
        <label htmlFor="">Email Id</label>
        <input
          type="email"
          id="email"
          value={email}
          onChange={(e) => setemail(e.target.value)}
        />
      </div>
      <div className="LoginInoutGroup">
        <label htmlFor="">Password ID</label>
        <input
          type="password"
          id="password"
          value={password}
          onChange={(e) => setpassword(e.target.value)}
        />
      </div>
      <button type="submit">LOGIN</button>
    </form>
  );
};

export default Login;
