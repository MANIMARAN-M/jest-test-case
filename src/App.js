import React, { useState } from "react";
import "./App.css";
import Login from "./Components/Login";

function App() {
  const [isLogged, setisLogged] = useState(false);
  return (
    <div data-testid="test-1" className="App">
      {isLogged ? (
        <div>Login Successfully</div>
      ) : (
        <Login data-testid="form-id" setisLogged={setisLogged} />
      )}
    </div>
  );
}

export default App;
