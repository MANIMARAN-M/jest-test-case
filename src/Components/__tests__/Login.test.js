import { render, screen, cleanup } from "@testing-library/react";
import App from "../../App";
import renderer from "react-test-renderer";
import Login from "../Login";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

// afterEach(cleanup);
describe("Login component tests", () => {
  const wrapper = shallow(<Login />);
  it("should have a btn component", () => {
    expect(wrapper.find("button")).toHaveLength(1);
    expect(wrapper.find("button").text()).toEqual("LOGIN");
  });
  it("should have input for email and password", () => {
    //Email and password input field should be present
    expect(wrapper.find("input#email")).toHaveLength(1);
    expect(wrapper.find("input#password")).toHaveLength(1);
    expect(wrapper.find("input")).toHaveLength(2);
  });
  it("should have an empty email and password state var", () => {
    //Optionally test to check if password and email are empty strings on setup
    expect(wrapper.find("#email").text()).toEqual("");
    expect(wrapper.find("#password").text()).toEqual("");
  });
  it("should take a Snapshop", () => {
    let tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
