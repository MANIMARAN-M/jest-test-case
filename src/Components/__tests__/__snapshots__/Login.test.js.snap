// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests should render correctly 1`] = `
<form
  className="LoginForm"
  id="Form"
  onSubmit={[Function]}
>
  <h1>
    signIn Form
  </h1>
  <div
    className="LoginInoutGroup"
  >
    <label
      htmlFor=""
    >
      Email Id
    </label>
    <input
      id="email"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="LoginInoutGroup"
  >
    <label
      htmlFor=""
    >
      Password ID
    </label>
    <input
      id="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </div>
  <button
    type="submit"
  >
    LOGIN
  </button>
</form>
`;

exports[`Login component tests should take a Snapshop 1`] = `
<form
  className="LoginForm"
  id="Form"
  onSubmit={[Function]}
>
  <h1>
    signIn Form
  </h1>
  <div
    className="LoginInoutGroup"
  >
    <label
      htmlFor=""
    >
      Email Id
    </label>
    <input
      id="email"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="LoginInoutGroup"
  >
    <label
      htmlFor=""
    >
      Password ID
    </label>
    <input
      id="password"
      onChange={[Function]}
      type="password"
      value=""
    />
  </div>
  <button
    type="submit"
  >
    LOGIN
  </button>
</form>
`;
